src/web/pages/reports/auditdetailspage.js

```diff
@@ -0,0 +1,713 @@
+
+import {useRouteMatch} from 'react-router-dom';
+
+import _ from 'gmp/locale';
+
+import logger from 'gmp/log';
+
+import Filter, {RESET_FILTER, RESULTS_FILTER_FILTER} from 'gmp/models/filter';
+import {isActive} from 'gmp/models/task';
+
+import {first} from 'gmp/utils/array';
+import {isDefined, hasValue} from 'gmp/utils/identity';
+
+import withDownload from 'web/components/form/withDownload';
+
+import Reload, {
+  NO_RELOAD,
+  USE_DEFAULT_RELOAD_INTERVAL_ACTIVE,
+} from 'web/components/loading/reload';
+
+import withDialogNotification from 'web/components/notification/withDialogNotifiaction'; // eslint-disable-line max-len
+
+import FilterProvider from 'web/entities/filterprovider';
+
+import DownloadReportDialog from 'web/pages/reports/downloadreportdialog';
+
+import {
+  loadAllEntities as loadFilters,
+  selector as filterSelector,
+} from 'web/store/entities/filters';
+
+import {
+  loadAllEntities as loadReportFormats,
+  selector as reportFormatsSelector,
+} from 'web/store/entities/reportformats';
+
+import {loadAuditReportWithThreshold} from 'web/store/entities/report/actions';
+import {auditReportSelector} from 'web/store/entities/report/selectors';
+
+import {
+  loadReportComposerDefaults,
+  renewSessionTimeout,
+  saveReportComposerDefaults,
+} from 'web/store/usersettings/actions';
+
+import {loadUserSettingDefaults} from 'web/store/usersettings/defaults/actions';
+import {getUserSettingsDefaults} from 'web/store/usersettings/defaults/selectors';
+import {getUserSettingsDefaultFilter} from 'web/store/usersettings/defaultfilters/selectors';
+
+import {
+  getReportComposerDefaults,
+  getUsername,
+} from 'web/store/usersettings/selectors';
+
+import {create_pem_certificate} from 'web/utils/cert';
+import compose from 'web/utils/compose';
+import {generateFilename} from 'web/utils/render';
+import PropTypes from 'web/utils/proptypes';
+import withGmp from 'web/utils/withGmp';
+
+import TargetComponent from '../targets/component';
+import PageTitle from 'web/components/layout/pagetitle';
+
+import Page from './auditdetailscontent';
+
+import FilterDialog from './detailsfilterdialog';
+import {pageFilter as setPageFilter} from 'web/store/pages/actions';
+import getPage from 'web/store/pages/selectors';
+import useGmp from 'web/utils/useGmp';
+
+const log = logger.getLogger('web.pages.auditreport.detailspage');
+
+const DEFAULT_FILTER = Filter.fromString(
+  'levels=hmlg rows=100 min_qod=70 first=1 compliance_levels=yniu sort=compliant',
+);
+
+export const AUDIT_REPORT_RESET_FILTER = RESET_FILTER.copy()
+  .setSortOrder('sort')
+  .setSortBy('compliant');
+
+const REPORT_FORMATS_FILTER = Filter.fromString('active=1 and trust=1 rows=-1');
+
+const getReportPageName = id => `report-${id}`;
+
+const getTarget = (entity = {}) => {
+  const {report = {}} = entity;
+  const {task = {}} = report;
+  return task.target;
+};
+
+const getFilter = (entity = {}) => {
+  const {report = {}} = entity;
+  return report.filter;
+};
```

Comment:
The whole function might be unnecessary because we can use the optional chaining (`?.`) operator nowadays. It was not available when we wrote all this code initially.